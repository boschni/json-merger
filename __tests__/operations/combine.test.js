const jsonMerger = require("../../dist");
const { testConfig } = require("../../__helpers__");

describe("when merging two objects and a source property has a $combine indicator", function () {
  test("it should combine the source property with the target", function () {
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: {
        $combine: [3, 3],
      },
    };

    const result = jsonMerger.mergeObjects([object1, object2], testConfig());

    expect(result).toMatchSnapshot();
  });

  test("it should combine if no defaultArrayMergeOperation specified", function () {
    const object1 = {
      a: [1, 2, 3],
    };

    const object2 = {
      a: [3, 3],
    };

    const result = jsonMerger.mergeObjects([object1, object2], testConfig());

    expect(result).toMatchSnapshot();
  });
});
