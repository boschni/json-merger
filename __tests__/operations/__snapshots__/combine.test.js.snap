// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects and a source property has a $combine indicator it should combine if no defaultArrayMergeOperation specified 1`] = `
"{
	\\"a\\": [
		3,
		3,
		3
	]
}"
`;

exports[`when merging two objects and a source property has a $combine indicator it should combine the source property with the target 1`] = `
"{
	\\"a\\": [
		3,
		3,
		3
	]
}"
`;
